basket/news/tasks.py

```diff
@@ -1314,122 +1132,12 @@ def amo_compress_categories(categories):
-    addon_data = {
-        "AMO_Category__c": amo_compress_categories(data["categories"]),
-        "AMO_Default_Language__c": data["default_locale"],
-        "AMO_GUID__c": data["guid"],
-        "AMO_Rating__c": data["ratings"]["average"],
-        "AMO_Slug__c": data["slug"],
-        "AMO_Status__c": data["status"],
-        "AMO_Type__c": data["type"],
-        "AMO_Update__c": data["last_updated"],
-        "Average_Daily_Users__c": data["average_daily_users"],
-        "Dev_Disabled__c": "Yes" if data["is_disabled"] else "No",
-        "AMO_Recommended__c": data["is_recommended"],
-    }
-    # check for possible None or empty values
-    if data["name"]:
-        addon_data["Name"] = data["name"]
-
-    # versions can be removed, so they should be removed if they are null
-    if data["current_version"]:
-        addon_data["AMO_Current_Version__c"] = data["current_version"]["version"]
-    else:
-        addon_data["AMO_Current_Version__c"] = ""
-
-    if data["latest_unlisted_version"]:
-        addon_data["AMO_Current_Version_Unlisted__c"] = data["latest_unlisted_version"][
-            "version"
-        ]
-    else:
-        addon_data["AMO_Current_Version_Unlisted__c"] = ""
-
-    sfdc.addon.upsert(f'AMO_AddOn_Id__c/{data["id"]}', addon_data)
-    addon_record = sfdc.addon.get_by_custom_id("AMO_AddOn_Id__c", data["id"])
-
-    # delete users no longer associated with the addon
-    existing_users = sfdc.sf.query(
-        sfapi.format_soql(
-            "SELECT Id, AMO_Contact_ID__c FROM DevAddOn__c WHERE AMO_AddOn_ID__c = {addon_id}",
-            addon_id=addon_record["Id"],
-        ),
-    )
-    user_ids_to_records = {
-        i["AMO_Contact_ID__c"]: i["Id"] for i in existing_users["records"]
-    }
-    existing_user_ids = set(user_ids_to_records.keys())
-    new_user_ids = {user["id"] for user in users}
-    if new_user_ids == existing_user_ids:
-        # no need to continue as no users have been added or removed
-        return
-
-    to_delete = existing_user_ids - new_user_ids
-    for delete_user_id in to_delete:
-        sfdc.dev_addon.delete(user_ids_to_records[delete_user_id])
-        amo_check_user_for_deletion(delete_user_id)
-
-    to_add = new_user_ids - existing_user_ids
-    if not to_add:
-        # no new users to add
-        return
-
-    for user in users:
-        if user["id"] not in to_add:
-            # record exists
-            continue
-
-        try:
-            sfdc.dev_addon.upsert(
-                f'ConcatenateAMOID__c/{user["amo_id"]}-{data["id"]}',
-                {
-                    "AMO_AddOn_ID__c": addon_record["Id"],
-                    "AMO_Contact_ID__c": user["id"],
-                },
-            )
-        except sfapi.SalesforceMalformedRequest as e:
-            if e.content[0]["errorCode"] == "DUPLICATE_VALUE":
-                # dupe error, so we don't need to do this again
-                pass
-            else:
-                raise e
+    pass
```

Comment:
I agree with the `pass` here, but also wonderning if a deprecation warning might be handy, to remind us that the AMO-related behaviour is a no-op?

Reply:
Sure, what would that look like exactly? I could raise an exception? I'll scan other parts of the code and maybe there is already a pattern for things like this.

Reply:
I think raising a `DeprecationWarning` might be the way forward, if I'm right that it won't actually blow up things, just shout loudly https://docs.python.org/3.9/library/exceptions.html#DeprecationWarning

```from warnings import warn

warn('This is a deprecated thing that needs a better message than this example', DeprecationWarning, stacklevel=2)
```

Reply:
I removed the AMO sync code